yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -408,31 +527,32 @@ func topoSort(dependencyMap map[string][]string) []string {
 // as foreign keys, populating their referenced table, column, and type information.
 // This is done after all DDL statements have been processed to ensure complete metadata is
 func (p *ParserIssueDetector) finalizeForeignKeyConstraints() {
-	for _, fk := range p.foreignKeyConstraints {
-		for i, localCol := range fk.ColumnNames {
-			if _, ok := p.columnMetadata[fk.TableName]; !ok {
-				p.columnMetadata[fk.TableName] = make(map[string]*ColumnMetadata)
-			}
-			meta, ok := p.columnMetadata[fk.TableName][localCol]
-			if !ok {
-				meta = &ColumnMetadata{}
-				p.columnMetadata[fk.TableName][localCol] = meta
-			}
-
-			meta.IsForeignKey = true
-			meta.ReferencedTable = fk.ReferencedTable
-			if i < len(fk.ReferencedColumns) {
-				refCol := fk.ReferencedColumns[i]
-				meta.ReferencedColumn = refCol
+	for tableName, tm := range p.tableMetadata {
```

Comment:
I'm hoping we have sufficient unit tests to verify that this refactor did not break anything?